R/render_all.R

```diff
@@ -0,0 +1,100 @@
+#' @title Function to render all protocols into a folder `publish/version_number`.
+#'
+#' @description This function renders all protocols (which have been merged to
+#' the master branch) into a folder `publish/version_number`, where
+#' `version_number` is a tag of the form `YYYY.NN` (year, number giving order of
+#'  release in a year).
+#'
+#'
+#' @param output_root A character string giving the root folder without version
+#' number. Default is "publish".
+#' @param version_number A character string of the form `YYYY.NN` (year, number
+#' giving order of release in a year).
+#'
+#' @return The rendered html files and associated files needed by the html file
+#' for all protocols will be written to subfolders of `publish/version_number`.
+#' The subfolder structure mirrors the structure of `src/thematic` and
+#' `src/project`.
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom assertthat assert_that is.string
+#' @importFrom fs dir_ls path path_rel
+#' @importFrom stringr str_detect
+#' @importFrom bookdown render_book
+#' @importFrom purrr pwalk
+#'
+#' @export
+#'
+#'
+#' @examples
+#' \dontrun{
+#' render_all(version_number = "2000.01")
+#'}
+
+
+render_all <- function(output_root = "publish",
+                       version_number = NULL) {
```

Comment:
don't set a default for `version_number`. Then you don't need the test of it being `NULL`.

Reply:
See 8131735